phalcon/di.zep

```diff
@@ -243,16 +265,15 @@ class Di implements DiInterface
 			}
 		}
 
-		/**
-		 * Pass the DI itself if the instance implements \Phalcon\Di\InjectionAwareInterface
-		 */
+		// Pass the DI to the instance if it implements \Phalcon\Di\InjectionAwareInterface
 		if typeof instance == "object" {
 			if instance instanceof InjectionAwareInterface {
 				instance->setDI(this);
 			}
 		}
 
-		if typeof eventsManager == "object" {
+		// Allows for post creation instance configuration through the "di:beforeServiceResolve" event.
+		if eventsManager !== null {
```

Comment:
The same here

Reply:
> let eventsManager = <ManagerInterface> this->_eventsManager;

@sergeyklay I was assuming that the type check `<ManagerInterface>` only allows it to be that type or null.  There is a check there (although maybe its only for IDE tools) and type checks when setting it.  I'm assuming that if it returns another object not of this type then it halts.  If my assumption is correct then my way will be marginally faster.